src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -247,10 +249,11 @@ private function dumpCallable($callable)
      */
     private function dumpValue($value)
     {
-        if ($value instanceof IteratorArgument) {
-            $value = array('=iterator' => $value->getValues());
-        } elseif ($value instanceof ClosureProxyArgument) {
-            $value = array('=closure_proxy' => $value->getValues());
+        if ($value instanceof IteratorArgument || $value instanceof ClosureProxyArgument) {
```

Comment:
we don't know the tag corresponding to other instances so I think it's better this way here.